### PR TITLE
listing of tasks

### PR DESCRIPTION
**Pull request by ChristianBeilschmidt**

### CODE REVIEW

**Comment by ChristianBeilschmidt** [comment]
[Tasks können in der UI verwaltet werden](https://trello.com/c/Y0u6Ybt0/1265-tasks-k%C3%B6nnen-in-der-ui-verwaltet-werden)

File: projects/core/src/lib/tasks/task-list/task-list.component.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    styleUrls: ['./task-list.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class TaskListComponent implements OnInit, AfterViewInit, OnDestroy {
+    taskStatusOptions: Array<TaskStatusType> = ['running', 'completed', 'aborted', 'failed'];
+    pageSize = 15; // current backend does not allow > 20 and we need one more for the next page check
+
+    displayedColumns: string[] = ['status', 'timeStarted', 'time', 'info', 'cleanUp', 'actions'];
+    tasks: Array<TaskStatusDict> = [];
+    filter?: TaskStatusType = undefined;
```
Comment (michaelmattig) [inline projects/core/src/lib/tasks/task-list/task-list.component.ts:20]: there is currently no way to reset the filter to no filter, once it is selected

File: projects/core/src/lib/tasks/task-list/task-list.component.html [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                <ng-container *ngIf="row.error">
+                    <pre>{{ row.error | json }}</pre>
+                </ng-container>
+            </td>
+        </ng-container>
+
+        <ng-container matColumnDef="timeStarted">
+            <th mat-header-cell *matHeaderCellDef>Started</th>
+            <td mat-cell *matCellDef="let row">
+                <ng-container *ngIf="row.timeStarted">{{ row.timeStarted }}</ng-container>
```
Comment (michaelmattig) [inline projects/core/src/lib/tasks/task-list/task-list.component.html:39]: would it make sense to render the time in local time?

File: projects/core/src/lib/tasks/task-list/task-list.component.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        } as PageEvent);
+    }
+
+    abortTask(taskId: UUID): void {
+        this.userService
+            .getSessionTokenForRequest()
+            .pipe(mergeMap((sessionToken) => this.backend.abortTask(sessionToken, taskId)))
+            .subscribe({
+                next: () => {
+                    this.refreshPage();
```
Comment (michaelmattig) [inline projects/core/src/lib/tasks/task-list/task-list.component.ts:113]: the abort can fail, e.g. if the task was finished in the meantime. We should print show the error (and refresh anyway)

File: projects/core/src/lib/tasks/task-list/task-list.component.html [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+                <ng-container *ngIf="row.timeTotal">{{ row.timeTotal }}</ng-container>
+                <ng-container *ngIf="!row.estimatedTimeRemaining && !row.timeTotal">-</ng-container>
+            </td>
+        </ng-container>
+
+        <ng-container matColumnDef="cleanUp">
+            <th mat-header-cell *matHeaderCellDef>Clean-up</th>
+            <td mat-cell *matCellDef="let row">
+                <ng-container *ngIf="row.cleanUp">
+                    <pre>{{ row.cleanUp | json }}</pre>
```
Comment (michaelmattig) [inline projects/core/src/lib/tasks/task-list/task-list.component.html:57]: we could render this a bit nicer, right? This, in contrast to the "info" column has a fixed format.

File: projects/core/src/lib/tasks/task-list/task-list.component.html (hunk 2 above)
Comment (ChristianBeilschmidt) [reply projects/core/src/lib/tasks/task-list/task-list.component.html]: [609d100](https://github.com/geo-engine/geoengine-ui/pull/617/commits/609d100194b12ae78fac167196b740aafb744bcc)

File: projects/core/src/lib/tasks/task-list/task-list.component.html (hunk 4 above)
Comment (ChristianBeilschmidt) [reply projects/core/src/lib/tasks/task-list/task-list.component.html]: [5f899de](https://github.com/geo-engine/geoengine-ui/pull/617/commits/5f899dee7b47c2caab6a821325eaeb2d5aac493d)

**Comment by michaelmattig** [comment]
bors merge

**Pull request merged**